sdk/java/pulumi/src/main/java/com/pulumi/resources/Resource.java

```diff
@@ -188,16 +190,7 @@ protected Resource(
             // the 'childResources' is a Synchronized Collection, so this is safe operation
             parentResource.childResources.add(this);
 
-            options.protect = options.protect || options.parent.protect; // TODO: is this logic good?
-
-            // Make a copy of the aliases array, and add to it any implicit aliases inherited from its parent
-            options.aliases = options.aliases == null ? new ArrayList<>() : copyNullableList(options.aliases);
```

Comment:
I'm not familiar with how this logic should work, but I've noticed we no longer set `options.aliases` here, is this right?